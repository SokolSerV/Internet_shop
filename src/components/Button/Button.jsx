import PropTypes from "prop-types";
import cn from "classnames";

import "./Button.scss";

export default function Button({
    type,
    backgroundColor,
    text,
    onClick,
    isDisabled,
}) {
    return (
        <button
            type={type}
            className={cn("button", { cursorDefault: isDisabled })}
            style={{ backgroundColor }}
            onClick={onClick}
            disabled={isDisabled}
        >
            {text}
        </button>
    );
}

Button.propTypes = {
    type: PropTypes.string,
    backgroundColor: PropTypes.string,
    text: PropTypes.string,
    onClick: PropTypes.func,
    isDisabled: PropTypes.bool,
};

Button.defaultProps = {
    type: "button",
    backgroundColor: "grey",
    text: "Кнопка",
    onClick: () => {},
    isDisabled: false,
};
