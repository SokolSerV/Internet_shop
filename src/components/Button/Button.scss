.button {
    border-radius: 7px;
    border-color: transparent;
    color: #ffffff;
    font-size: 15px;
    font-weight: 700;
    line-height: 15px;
    padding: 7px 7px;
    width: max-content;
    min-width: 100px;
    cursor: pointer;

    &.cursorDefault {
        cursor: default;
    }
}
