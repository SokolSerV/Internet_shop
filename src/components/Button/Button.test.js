import { fireEvent, render, screen } from "@testing-library/react";

import Button from "./Button";

describe("Button component Smoke test (Snapshot test)", () => {
    test("should Button render", () => {
        const { asFragment } = render(
            <Button
                type="reset"
                backgroundColor="red"
                text="Test text"
                isDisabled={true}
            />
        );

        expect(asFragment()).toMatchSnapshot();
    });

    describe("Button ckick", () => {
        const onClickMock = jest.fn();

        test("should onClick triggering by button click", () => {
            render(<Button onClick={onClickMock} text="Test text" />);
            // eslint-disable-next-line testing-library/no-debugging-utils
            screen.debug();

            const btn = screen.getByRole("button");
            fireEvent.click(btn);

            expect(onClickMock).toHaveBeenCalled();
        });
    });
});
