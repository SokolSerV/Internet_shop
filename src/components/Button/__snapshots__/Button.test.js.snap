// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Smoke test (Snapshot test) should Button render 1`] = `
<DocumentFragment>
  <button
    class="button cursorDefault"
    disabled=""
    style="background-color: red;"
    type="reset"
  >
    Test text
  </button>
</DocumentFragment>
`;
